packages/lit-html/src/directives/cache.ts

```diff
@@ -48,12 +54,12 @@ class CacheDirective extends Directive {
       if (cachedContainerPart === undefined) {
         const fragment = document.createDocumentFragment();
         cachedContainerPart = render(nothing, fragment);
+        cachedContainerPart.setConnected(false);
```

Comment:
This is set earlier here because `insertPart` synchronizes the connection state?

Reply:
Right, previously we changed the connection state on the `ChildPart` that we inserted and moved between the cache and the container parts as we moved them. Now, `cachedContainerPart` is a `RootChildPart`, so we set it to "disconnected" once, and each time we move a part in and out, its connection state will be synced to the root part it is in.